datafusion/functions-nested/src/array_has.rs

```diff
@@ -323,3 +316,97 @@ fn general_array_has_dispatch<O: OffsetSizeTrait>(
     }
     Ok(Arc::new(boolean_builder.finish()))
 }
+
+/// Public function for internal benchmark, avoid to use it in production
+pub fn array_has_dispatch<O: OffsetSizeTrait>(
+    haystack: &ArrayRef,
+    needle: &ArrayRef,
+) -> Result<ArrayRef> {
+    let haystack = as_generic_list_array::<O>(haystack)?;
+    match needle.data_type() {
+        DataType::Utf8 => array_has_string_internal::<O, i32>(haystack, needle),
+        DataType::LargeUtf8 => array_has_string_internal::<O, i64>(haystack, needle),
+        DataType::Utf8View => array_has_string_view_internal::<O>(haystack, needle),
+        _ => general_array_has::<O>(haystack, needle),
+    }
+}
+
+fn array_has_string_internal<O: OffsetSizeTrait, S: OffsetSizeTrait>(
+    array: &GenericListArray<O>,
+    needle: &ArrayRef,
+) -> Result<ArrayRef> {
+    let mut boolean_builder = BooleanArray::builder(array.len());
+    let needle_array = needle.as_string::<S>();
+    for (arr, element) in array.iter().zip(needle_array.iter()) {
+        match (arr, element) {
+            (Some(arr), Some(element)) => {
+                let string_arr = arr.as_string::<S>();
+                let mut res = false;
+                for sub_arr in string_arr.iter().flatten() {
+                    if sub_arr == element {
+                        res = true;
+                        break;
+                    }
+                }
+                boolean_builder.append_value(res);
+            }
+            (_, _) => {
+                boolean_builder.append_null();
+            }
+        }
+    }
+
+    Ok(Arc::new(boolean_builder.finish()))
+}
+
+fn array_has_string_view_internal<O: OffsetSizeTrait>(
```

Comment:
I prefer not to macro these two functions

Reply:
Use StringArrayType for needle instead?

Reply:
I doesn't find this helpful, since I need to get the inner data in array too. That ends up declaring two functions with `StringArrayType`